### PR TITLE
[macos][msbuild] Cache actool results like XI

### PR DESCRIPTION
**Pull request by chamons**
- https://github.com/xamarin/xamarin-macios/issues/3584
- Prevents a 2 second pause on every single launch

### CODE REVIEW

File: msbuild/Xamarin.Mac.Tasks/Xamarin.Mac.Common.targets [hunk 1]
```diff
@@ -558,12 +566,40 @@ Copyright (C) 2014 Xamarin. All rights reserved.
 		</CreateItem>
 	</Target>
 
-	<Target Name="_CompileImageAssets" DependsOnTargets="_DetectAppManifest;_DetectSdkLocations;_CoreCompileImageAssets" />
+	<Target Name="_CompileImageAssets" DependsOnTargets="_DetectAppManifest;_DetectSdkLocations;_BeforeCoreCompileImageAssets;_ReadCompileImageAssets;_CoreCompileImageAssets" />
```
Comment (chamons) [inline msbuild/Xamarin.Mac.Tasks/Xamarin.Mac.Common.targets]: All of this logic is ported straight from XI msbuild.

File: tests/common/mac/ProjectTestHelpers.cs [hunk 2]
```diff
@@ -306,6 +309,24 @@ public static string GenerateEXEProject (UnifiedTestConfig config)
...
+    <ImageAsset Include=""Assets.xcassets\AppIcon.appiconset\AppIcon-128.png"" />
+    <ImageAsset Include=""Assets.xcassets\AppIcon.appiconset\AppIcon-128%402x.png"" />
+    <ImageAsset Include=""Assets.xcassets\AppIcon.appiconset\AppIcon-16.png"" />
+    <ImageAsset Include=""Assets.xcassets\AppIcon.appiconset\AppIcon-16%402x.png"" />
+    <ImageAsset Include=""Assets.xcassets\AppIcon.appiconset\AppIcon-256%402x.png"" />
+    <ImageAsset Include=""Assets.xcassets\AppIcon.appiconset\AppIcon-32.png"" />
+    <ImageAsset Include=""Assets.xcassets\AppIcon.appiconset\AppIcon-32%402x.png"" />
+    <ImageAsset Include=""Assets.xcassets\Contents.json"" />
+  </ItemGroup>";
+				// HACK - Should process using CopyFileWithSubstitutions
```
Comment (chamons) [inline tests/common/mac/ProjectTestHelpers.cs]: I could refactor multiple places for this but wanted to keep diff cleaner.

**Comment by chamons** [comment]
There are a bunch of caching XI does that XM does not, but this one is the worst offender, since new XM project hits this 2 second penalty every single launch. It's really annoying.

**Comment by chamons** [comment]
Also, in addition to the "let's not be silly" autotest, I also manually tested this by adding\removing an image to the catalog and checking that it updates the final bundle, and then rebuilds don't change it.

I could port that to an auto test if needed, but it's a bit overkill given the rest of our msbuild test coverage.

**Comment by filipnavara** [comment]
Thanks for the early christmas! I will give it a try on Monday, but I have high hopes that it could save us 30+ seconds on every build because it was triggering whole tree of dependencies to be rebuilt every single time.

**Comment by chamons** [comment]
I disabled the test due to everyone most not favorite issue - https://github.com/xamarin/xamarin-macios/issues/4110 :(

**Comment by filipnavara** [comment]
I tested it on our repository and the improvements for iterative builds are enormous. Analysis of the structured build output shows that none of the dependent libraries are rebuilt now, which was previously the case.

**Comment by spouliot** [comment]
@chamons that looks relatively safe to backport to 15.9 considering it's similar to XI and we have Filip confirmation that it works (better) for him. Please backport if you agree :)

**Pull request merged**